src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -441,6 +444,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addSuperinterface(superinterface: KClass<*>)
         = addSuperinterface(superinterface.asTypeName())
 
+    fun addSuperinterface(superinterface: ParameterSpec) = apply {
```

Comment:
You won't be able to re-use `ParameterSpec` not only because of the aforementioned different type situation, but also because you can do things like
```kotlin
interface B
class Foo: B
class A : B by Foo()
```